src/mongocxx/private/search_index_view.hh

```diff
@@ -0,0 +1,192 @@
+#pragma once
+
+#include <vector>
+
+#include <bsoncxx/builder/basic/array.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/document/view_or_value.hpp>
+#include <bsoncxx/string/to_string.hpp>
+#include <bsoncxx/string/view_or_value.hpp>
+#include <bsoncxx/types/bson_value/view.hpp>
+#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/private/append_aggregate_options.hh>
+#include <mongocxx/private/client_session.hh>
+#include <mongocxx/private/libbson.hh>
+#include <mongocxx/private/libmongoc.hh>
+
+#include <mongocxx/config/private/prelude.hh>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+using bsoncxx::builder::basic::kvp;
+using bsoncxx::builder::basic::make_document;
+
+class search_index_view::impl {
+   public:
+    impl(mongoc_collection_t* collection, mongoc_client_t* client)
+        : _coll{collection}, _client{client} {}
+
+    impl(const impl& i) = default;
+
+    impl(impl&& i) = default;
+
+    ~impl() = default;
+
+    impl& operator=(const impl& i) = default;
```

Comment:
```suggestion
    ~impl() = default;
    impl(impl&& i) = delete;
    impl& operator=(impl&& i) = delete;
    impl(const impl& i) = delete;
    impl& operator=(const impl& i) = delete;
```
All `= default` -> none necessary. That being said, would recommend deleting copy/move special members as the `impl` class is not expected to be directly copied or moved.